pkg/util/resizefs/resizefs_linux.go

```diff
@@ -84,3 +100,19 @@ func (resizefs *ResizeFs) xfsResize(deviceMountPath string) (bool, error) {
 	resizeError := fmt.Errorf("resize of device %s failed: %v. xfs_growfs output: %s", deviceMountPath, err, string(output))
 	return false, resizeError
 }
+
+// findBlockDeviceRescanPath Find the underlaying disk for a linked path such as /dev/disk/by-path/XXXX or /dev/mapper/XXXX
+// will return /sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:1/0:0:1:0/rescan
+func findBlockDeviceRescanPath(path string) (string, error) {
+	devicePath, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return "", err
+	}
+	// if path /dev/hdX split into "", "dev", "hdX" then we will
+	// return just the last part
+	parts := strings.Split(devicePath, "/")
+	if len(parts) == 3 && strings.HasPrefix(parts[1], "dev") {
+		return filepath.EvalSymlinks(filepath.Join("/sys/block", parts[2], "device", "rescan"))
```

Comment:
This should be kind of an opt-in. Not all volume types support rescanning and making this as core part of resizing workload will break those plugins. For example on EBS:

```
# pwd
/sys/block/xvdbd/device
# ls
block  devtype  driver  modalias  nodename  power  subsystem  uevent
```

I am also wondering which in-tree plugins are targetted by this change? Is this change for CSI plugins that are vendoring k8s and using resize helper defined here?  If yes - we may want to do this as an optional flag to `Resize` function rather than making it a default.

Reply:
It won't break, but show a log message. Though making this an option is a good idea.

Reply:
I think passing a boolean flag called `rescanDevice` to `Resize` function makes sense. All of existing in-tree drivers will simply pass `false` but if any driver needs it they can opt-in.

Reply:
Implemented